common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1197,7 +1205,11 @@ def _block_matches_all(block_data):
             settings['children'] = qualifiers.pop('children')
         for block_id, value in course.structure['blocks'].iteritems():
             if _block_matches_all(value):
-                items.append(block_id)
+                if not include_orphans:
+                    if self.has_path_to_root(block_id, course) or block_id.type in _DETACHED_CATEGORIES:
```

Comment:
if we checked `block_id.type in _DETACHED_CATEGORIES` first, we'll get a little performance boost, since if that passed, we wouldn't have to check `self.has_path_to_root(block_id, course)`